### ISSUE TITLE
Potential for federated wiki on sandstorm.io?

### ISSUE THREAD

**Issue by coevolving** [WardCunningham/Smallest-Federated-Wiki#447]
One of the deterrents for federated wiki is that it isn't a one-button install.

Have you seen https://sandstorm.io/ ?  Given my experience with the Openshift Quickstart, this path might be even easier for a newbie to get up.

I came to this because I was looking for a collaborative SVG editor, and saw that draw.io is included.  There's even an implementation for Apache Wave (nee Google Wave)!

Maybe this is a platform worth contributing to.  (While you're at it, maybe the farm version of federated wiki could be the default installation!)

P.S.  The other platform that is widespread is cPanel.  When I get some more time, I really should try installing the Ruby version of federated wiki on my usual domain.  When I tried last year, the Ruby was downlevel, which made the Quickstart a faster alternative.

**Comment by paul90** [comment]
sandstorm.io was mentioned in yesterday's hangout

**Comment by WardCunningham** [comment]
The ruby version hasn't been maintained and should be avoided. (I still run the ruby version on fed.wiki.org but patch in revised javascript by hand.)

**Comment by coevolving** [comment]
@paul90 On yesterday's hangout, was there an assessment or evaluation of federated wiki with sandstorm.io?  Favourable or unfavourable?  (Sorry, I couldn't be in two places at one time).

@WardCunningham Thanks for the reminder to avoid the Ruby version.  It changes the options for implementation.

**Comment by paul90** [comment]
Favourable, though I'm not sure how much depth it has been looked at yet, or if anybody has it up and running yet - from a cursory glance it looks interesting.

**Comment by opn** [comment]
Yes. I looked at this in some detail yesterday - though not to the extent of downloading and running code. I'm pretty keen on this direction at first site. It certainly fits my coding style and my security preferences and offers a great scope with regard to giving students their own work and coding sandboxes. I contacted them yesterday about getting involved at Chaos communications Camp - they seem interested. @paul90 you want to come camping ?

**Comment by paul90** [comment]
@opn I don't think so, I only ever wilderness camped...

**Comment by timothyfcook** [comment]
Wow, I was just browsing Fedwiki install instructions and Sandstorm and thinking the same thing and came here to create an issue.

Lo and behold...

This sort of one-button install is exactly what FedWiki needs. Maybe a bounty is in order...

**Comment by timothyfcook** [comment]
https://www.bountysource.com/issues/23026315-potential-for-federated-wiki-on-sandstorm-io

**Comment by almereyda** [comment]
I'd happy to have a look at this task, as we already have Docker containers
for Federated Wiki. We just need to design them in their preferred way.
Do we already know about documentation about how to integrate applications
into Sandstorm, especially regarding its autoupdate feature?

**Comment by timothyfcook** [comment]
Awesome. I know a lot of people interested in FedWiki, but don't have the time or expertise to get it stood up.

**Comment by pierreozoux** [comment]
@almereyda thanks for the ping.
I don't integrate images on IndieHosters until a user asks me to. #lean :)

**Comment by almereyda** [comment]
We'll first have to go through https://docs.sandstorm.io/en/latest/vagrant-spk/packaging-tutorial/ step by step.

**Comment by paul90** [comment]
From memory, you will at the very least need to create a custom version of the wiki-node package - sandstorm appears to put the package at root which causes `path.join` to fail with not a string error. 

Also you will need to move the data directory into `/var` - as that appears to be the only place that is mounted r/w.

Not too sure if farm mode, with dynamic creation of wiki's, will work. While it is documented how to point a single domain at a grain, I very much doubt that it will work with a wildcard.

As far as I can see sites running on Sandstorm are only accessable over HTTPS - so you will need to proxy Sandstorm to make it available without HTTPS.
